scylla/tests/integration/statements/coordinator.rs

```diff
@@ -106,3 +111,141 @@ async fn test_enforce_non_existent_request_coordinator() {
     let result = session.query_unpaged(statement, ()).await;
     assert_matches!(result, Err(ExecutionError::EmptyPlan))
 }
+
+/// Checks that if a node is enforced as the coordinator of a request, the [Coordinator] struct
+/// exposed on the request result (`QueryResult` and `QueryPager`) contains that `Node`.
+#[tokio::test]
+async fn test_exposed_request_coordinator() {
+    setup_tracing();
+    let session = create_new_session_builder().build().await.unwrap();
+
+    // Create schema for batches.
+    {
+        let ks = unique_keyspace_name();
+        session
+            .query_unpaged(
+                format!(
+                    "CREATE KEYSPACE IF NOT EXISTS {ks} WITH replication = {{'class': 'NetworkTopologyStrategy', 'replication_factor': 1}}"
+                ),
+                (),
+            )
+            .await
+            .unwrap();
+        session.use_keyspace(ks, true).await.unwrap();
+        session
+            .query_unpaged("CREATE TABLE IF NOT EXISTS test (a int PRIMARY KEY)", ())
+            .await
+            .unwrap();
+    }
+    let mut batch = Batch::new_with_statements(
+        BatchType::Logged,
+        vec![BatchStatement::Query(Statement::new(
+            "INSERT INTO test (a) VALUES (?)",
+        ))],
+    );
+    let mut next_pk = {
+        let mut pk = -1;
+        move || {
+            pk += 1;
+            pk
+        }
+    };
+
```

Comment:
What is this for? Unless I misunderstood something, this test would work just as well with hardcoded values.

Reply:
You mean it would work by inserting only one row?

Reply:
This is to generate unique rows, so that each new batch inserts another new row.

Reply:
But why? Contents of the table are not queried anywhere, so it does not matter if you only insert one row, or unique rows.

Reply:
Done.